### PR TITLE
`<host_version>` syntax

### PR DESCRIPTION
**Pull request by franramirez688**
Docs for https://github.com/conan-io/conan/pull/13712

### CODE REVIEW

**Review by czoido — APPROVED** [review]

File: reference/conanfile/methods/build_requirements.rst [hunk 1]
```diff
@@ -27,6 +27,45 @@ The ``tool_requires`` is equivalent to ``requires()`` with the following traits:
 - ``headers=False`` A tool requirement does not have headers.
 - ``libs=False``: A tool requirement does not have libraries to be linked by the consumer (if it had libraries they would be in the "build" context and could be incompatible with the "host" context of the consumer package). 
 
+<host_version>
+**************
+
+.. include:: ../../../common/experimental_warning.inc
+
+This syntax is useful when you're using the same package recipe as a *requires* and as a *tool_requires* and you want to avoid
+conflicting downstream if any user decides to override the original *requires* version, i.e., the user could end up with
```
Comment (jcar87) [inline reference/conanfile/methods/build_requirements.rst:36]:
```suggestion
conflicting downstream if any user decides to override the original *requires* version in the *host* context, i.e., the user could end up with
```

File: reference/conanfile/methods/build_requirements.rst [hunk 2]
```diff
@@ -27,6 +27,45 @@ The ``tool_requires`` is equivalent to ``requires()`` with the following traits:
...
+two different versions in host and build context of the same dependency.
```
Comment (jcar87) [inline reference/conanfile/methods/build_requirements.rst:37]: I would consider adding something along the lines of:

"The `<host_version>` specifier allows us to ensure that the version resolved for the `tool_requires` always matches the one for the host requirement.

File: reference/conanfile/methods/build_requirements.rst [hunk 3]
```diff
@@ -27,6 +27,45 @@ The ``tool_requires`` is equivalent to ``requires()`` with the following traits:
...
 - ``libs=False``: A tool requirement does not have libraries to be linked by the consumer (if it had libraries they would be in the "build" context and could be incompatible with the "host" context of the consumer package). 
 
+<host_version>
+**************
+
+.. include:: ../../../common/experimental_warning.inc
+
+This syntax is useful when you're using the same package recipe as a *requires* and as a *tool_requires* and you want to avoid
+conflicting downstream if any user decides to override the original *requires* version in the *host* context, i.e., the user could end up with
+two different versions in host and build context of the same dependency.
```
Comment (RubenRBS) [inline reference/conanfile/methods/build_requirements.rst:37]:
```suggestion
two different versions in the host and build contexts of the same dependency.
```

File: reference/conanfile/methods/build_requirements.rst [hunk 4]
```diff
@@ -27,6 +27,45 @@ The ``tool_requires`` is equivalent to ``requires()`` with the following traits:
...
+
+    class mylibRecipe(ConanFile):
+        name = "mylib"
+        version = "0.1"
+        def requirements(self):
+            self.requires("protobuf/3.18.1")
+        def build_requirements(self):
+            self.tool_requires("protobuf/<host_version>")
+
+Then, if any user wants to use my *mylib/0.1*, but another version of protobuf, there shouldn't be any problem overriding it:
```
Comment (RubenRBS) [inline reference/conanfile/methods/build_requirements.rst:56]:
```suggestion
Then, if any user wants to use *mylib/0.1*, but another version of *protobuf*, there shouldn't be any problems overriding it:
```

File: reference/conanfile/methods/build_requirements.rst [hunk 5]
```diff
@@ -27,6 +27,45 @@ The ``tool_requires`` is equivalent to ``requires()`` with the following traits:
...
+    from conan import ConanFile
+
+    class myappRecipe(ConanFile):
+        name = "myapp"
+        version = "0.1"
+        def requirements(self):
+            self.requires("mylib/0.1")
+            self.requires("protobuf/3.21.9", override=True)
+
+The ``<host_version>`` defined upstream is ensuring that the host and build context are using the same version of that requirement.
```
Comment (RubenRBS) [inline reference/conanfile/methods/build_requirements.rst:69]:
```suggestion
The ``<host_version>`` defined upstream is ensuring that the host and build contexts are using the same version of that requirement.
```

**Review by RubenRBS — APPROVED** [review]
Nice!

**Review by memsharded — APPROVED** [review]

**Pull request merged**